apis/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -88,14 +95,44 @@ type ForeignClusterStatus struct {
 	// +kubebuilder:validation:Optional
 	TenantControlNamespace TenantControlNamespace `json:"tenantControlNamespace"`
 
-	Outgoing Outgoing `json:"outgoing,omitempty"`
-	Incoming Incoming `json:"incoming,omitempty"`
-	// It stores most important network statuses.
-	Network Network `json:"network,omitempty"`
-	// Authentication status.
-	// +kubebuilder:validation:Enum="Pending";"Accepted";"Refused";"EmptyRefused"
-	// +kubebuilder:default="Pending"
-	AuthStatus discovery.AuthStatus `json:"authStatus,omitempty"`
+	// PeeringConditions contains the conditions over the peering related to this
+	// ForeignCluster.
+	PeeringConditions []PeeringCondition `json:"peeringConditions,omitempty"`
+}
+
+// PeeringConditionType represents different conditions that a peering could assume.
+type PeeringConditionType string
+
+// These are valid conditions of a peering.
+const (
+	// OutgoingPeeringCondition informs users about the outgoing peering status.
+	OutgoingPeeringCondition PeeringConditionType = "OutgoingPeeringCondition"
+	// IncomingPeeringCondition informs users about the incoming peering status.
+	IncomingPeeringCondition PeeringConditionType = "IncomingPeeringCondition"
+	// LocalNetworkConfigPeeringCondition informs users about the local NetworkConfig status.
+	LocalNetworkConfigPeeringCondition PeeringConditionType = "LocalNetworkConfigPeeringCondition"
+	// RemoteNetworkConfigPeeringCondition informs users about the remote NetworkConfig status.
+	RemoteNetworkConfigPeeringCondition PeeringConditionType = "RemoteNetworkConfigPeeringCondition"
+	// TunnelEndpointPeeringCondition informs users about the TunnelEndpoint status.
+	TunnelEndpointPeeringCondition PeeringConditionType = "TunnelEndpointPeeringCondition"
```

Comment:
Was wondering whether it was possible to merge these three under a more generic NetworkUnavailable condition. As a user, I do not know about the internal details of liqo (networkconfig, tunnelendpoint, ...), but I am interested in knowing whether the network interconnection works or not. Then, in case of problems, the details could be provided in the reason/message fields, to know where to look for troubleshooting.